src/european_values/generative_training.py

```diff
@@ -68,39 +72,82 @@ def train_generative_model(
     n_components_range = [1, 2, 3, 4, 5, 10, 15, 20, 25, 30]
     n_components_range = [n for n in n_components_range if n <= max_components]
 
-    # Create GMM instance once
-    gmm = GaussianMixture(random_state=seed)
-
     bic_scores = {}
+
```

Comment:
Nit: Removing the newline indicates that the `bic_scores = {}` line belongs to the code block afterwards.

```suggestion
```